docs/en/advanced_config/bootloader_secure_boot.md

```diff
@@ -0,0 +1,323 @@
+
+## Troubleshooting
+
+### Common Issues
+
+#### Firmware won't boot after enabling secure boot
+
+- Ensure the firmware is properly signed
+- Verify the TOC structure is correct
+- Check that the key index in TOC matches the keystore
+
+#### Signature verification always fails
+
+- Confirm public key is correctly installed in keystore
+- Verify the signature was created with the matching private key
+- Check that the signature algorithm matches (e.g., ED25519)
+- Ensure the signature covers the correct data range
+
+#### Device appears bricked
+
+- The bootloader should always be accessible
+- Use a debug probe to reprogram the bootloader if needed
+- Check that BOOTLOADER_USE_SECURITY is defined correctly
+
+### Debug Information
+
+To debug secure boot issues:
+
+1. Enable bootloader debug output (if supported by your board)
+2. Use a debug probe to examine memory and execution flow
+3. Verify TOC parsing with debug tools
+4. Check return values from `verify_app()` function
+
+## How It Works
+
+### Boot Process
+
+1. **Bootloader starts**: The secure bootloader runs first after power-on or reset.
+2. **TOC validation**: Locates and validates the Firmware Table of Contents (TOC) structure in flash.
+3. **Signature verification**: Verifies the cryptographic signature of the firmware using stored keys.
+4. **Boot decision**:
+   - If verification succeeds, the firmware is executed.
+   - If verification fails, the bootloader remains active and waits for a valid firmware upload.
+
+### Firmware Table of Contents (TOC)
+
+The firmware TOC is a data structure embedded in the firmware that describes:
+
+- Firmware components and their locations in flash memory
+- Signature information for each component
+- Encryption settings (if used)
+- Boot flags indicating which component should be executed
+
+The TOC format is defined in `src/include/image_toc.h`:
+
+```c
+typedef struct image_toc_entry {
+    unsigned char name[4];      // Name of the section
+    const void *start;          // Start address in flash
+    const void *end;            // End address in flash
+    void *target;               // Copy target address (for RAM execution)
+    uint8_t signature_idx;      // Index to signature in TOC
+    uint8_t signature_key;      // Key index for signature verification
+    uint8_t encryption_key;     // Key index for encryption (future)
+    uint8_t flags1;             // Control flags
+    uint32_t reserved;          // Reserved for future use
+} image_toc_entry_t;
+```
+
+### Supported Cryptographic Algorithms
+
+PX4's crypto backend supports multiple algorithms (defined in `platforms/common/include/px4_platform_common/crypto_algorithms.h`):
+
+- **ED25519**: EdDSA signatures using Curve25519 (recommended for signatures)
+- **RSA-OAEP**: RSA with Optimal Asymmetric Encryption Padding
+- **XCHACHA20**: Stream cipher for encryption (future use)
+- **AES**: Block cipher for encryption (future use)
```

Comment:
Is this accurate? Which can we use for signing? My understanding is only ED25519 - the rest are defined but not implemented anywhere - right?